content/en/docs/tasks/run-application/access-api-from-pod.md

```diff
@@ -42,10 +42,11 @@ securely with the API server.
 
 ### Directly accessing the REST API
 
-While running in a Pod, the Kubernetes apiserver is accessible via a Service named
-`kubernetes` in the `default` namespace. Therefore, Pods can use the
-`kubernetes.default.svc` hostname to query the API server. Official client libraries
-do this automatically.
+While running in a Pod, the Kubernetes API server is referenced by the environment
+variables `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT`. As a convenience,
+the API server may also be referenced via a Service named `kubernetes` in the `default`
+namespace. Therefore, Pods can use the `kubernetes.default.svc` hostname to query the
+API server.
```

Comment:
Is this true:
```suggestion
While running in a Pod, your container can work our a URL for Kubernetes API server by
fetching the two  environment variables `KUBERNETES_SERVICE_HOST` and
`KUBERNETES_SERVICE_PORT_HTTPS`.

For apps that have access to the `default` namespace, the API server may also be referenced
via a Service named `kubernetes` in the `default` namespace. This means that Pods can use 
the `kubernetes.default.svc` hostname to create a HTTPS URL for the API server.

{{< note >}}
Kubernetes does not guarantee that the API server has a valid certificate for
the hostname  `kubernetes.default.svc`;
however, the control plane **is** expected to present a valid certificate for the
hostname or IP address that `$KUBERNETES_SERVICE_HOST` represents.
{{< /note >}}
```

Reply:
> For apps that have access to the `default` namespace

Is this true? My understanding is that DNS discoverability isn't partitioned. So everything in the cluster should be able to find this service regardless of whether the pod has access to read services in the `default` namespace.

Reply:
What I actually wanted to convey is that all clients can expect to resolve `kubernetes.default.svc`, and that additionally clients that have read access to the `kubernetes` Service (and its EndpointSlices) in NS `default` can also fetch the endpoint info another way. Probably too much detail.

Feel free to reword to suit @olix0r .

Reply:
Rereading your suggestion, I see what you mean now.

Practically, it seems unlikely that a pod that already has access to the API server will need to read the Service/Endpoints--this information is almost always used when bootstrapping a client that does not yet have API access.

I've  kept your note about the TLS certificate, however.